### ISSUE TITLE
Allow platform-specific startup options

### ISSUE THREAD

**Issue by peakschris** [bazelbuild/bazel#22763]
### Description of the feature request:

Where a startup option is platform specific, currently there is no way to handle it in a single .bazelrc file that supports both platforms.

This would be desired:
```
startup:windows --host_jvm_args=-Djavax.net.ssl.trustStoreType=Windows-ROOT
startup:linux --host_jvm_args=-Djavax.net.ssl.trustStore=/etc/pki/java/cacerts
```

A similar syntax and set of built-in platforms to those supported by `--enable_platform_specific_config` would seem reasonable. I am **not** suggesting that we support the --config command line for startup.

### Which category does this issue belong to?

Configurability

### What underlying problem are you trying to solve with this feature?

A single .bazelrc file that can build multiple platforms without needing to 'configure' the file prior to first build.

### Which operating system are you running Bazel on?

windows

### What is the output of `bazel info release`?

7.2.0

**Comment by aranguyen** [comment]
@peakschris and would this be for host platforms? [`--enable_platform_specific_config`](https://bazel.build/run/bazelrc#enable_platform_specific_config) makes the selection based on the host platform

**Comment by peakschris** [comment]
Yes, host platform I believe makes sense.

**Comment by Spectre0911** [comment]
@peakschris, currently attempting to do a very similar thing, did you manage to find a nice work around to this?

**Comment by peakschris** [comment]
Unfortunately not. We have a configure script for windows and linux that copies a platform-specific .bazelrc file into '%workspace%\.bazelrc.plat' and this is referenced by .bazelrc. I don't think there is any alternative at this time to avoid a configure script.

**Comment by Spectre0911** [comment]
@peakschris thanks for the response, hopefully this feature is worked on as it would certainly simplify things!

**Comment by katre** [comment]
I agree this sounds useful, and is in line with existing `--enable_platform_specific_config` usage.

Unfortunately, we can't prioritize this now, but if someone wants to work on a solution we can work on reviewing and merging it.

**Comment by keith** [comment]
https://github.com/bazelbuild/bazel/pull/26882

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 9.1.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/9.1.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=9.1.0rc1`. Thanks!
